dataflow_transfer/dataflow_transfer.py

```diff
@@ -1,28 +1,75 @@
-import yaml
 import logging
 import os
 
+from dataflow_transfer.run_classes.illumina_runs import (
+    NovaSeqXPlusRun,
+    NextSeqRun,
+    MiSeqRun,
+)
+from dataflow_transfer.run_classes.ont_runs import PromethIONRun, MinIONRun
+from dataflow_transfer.run_classes.element_runs import AvitiRun
+
 logger = logging.getLogger(__name__)
 
 
+def get_run_object(run_dir, sequencer, config):
+    if sequencer == "NovaSeqXPlus":
```

Comment:
Couldn't we try a registry here? https://www.abhik.xyz/articles/registry-pattern

Reply:
Yes, we should consider if a design pattern would be useful here. The current implementation resembles a factory method (though I've only scratched the surface of design patterns, so please correct me if I'm wrong). What would be the benefit of switching to a registry pattern?